tmt/utils.py

```diff
@@ -555,6 +555,23 @@ def fail(self, message: str, shift: int = 0) -> None:
         """ Show a red failure message on info level, send to stderr """
         self.info('fail', message, color='red', shift=shift, err=True)
 
+    def _command_verbose_logger(
+            self,
+            key: str,
+            value: Optional[str] = None,
+            color: Optional[str] = None,
+            shift: int = 2,
+            level: int = 3,
+            err: bool = False) -> None:
+        """
+        Reports the executed command in verbose mode.
+
+        This is a tailored verbose() function used for command logging where
+        default parameters are adjusted (to preserve the function type).
+        """
+        self.verbose(key=key, value=value, color=color, shift=2,
+                     level=3, err=err)
```

Comment:
I did some quick experiments and see the following output:

```
/plans/example
summary: Basic smoke test
    prepare
        how: shell
        order: 50
        overview: 1 script found
        script: dnf install -y tree
        cmd: set -eo pipefail; dnf install -y tree
                out: Last metadata expiration check: 0:20:03 ago on Wed 09 Nov 2022 01:02:06 PM UTC.
                out: Package tree-2.0.2-1.fc36.x86_64 is already installed.
                out: Dependencies resolved.
                out: Nothing to do.
                out: Complete!
    
        how: install
        summary: Install required packages
        name: requires
        order: 70
        cmd: stat /run/ostree-booted
                err: stat: cannot statx '/run/ostree-booted': No such file or directory
        cmd: rpm -q dnf
                out: dnf-4.11.1-2.fc36.noarch
        cmd: whoami
                out: root
        package: 1 package requested
            beakerlib
        cmd: rpm -q --whatprovides "beakerlib" || dnf install -y "beakerlib"
                out: beakerlib-1.29.3-1.fc36.noarch
    
        summary: 2 preparations applied
```

It seems that the shift level will need some adjustments. Above the `cmd` and `out` lines are not aligned. They should have the same indent, or, may be command output should go one level deeper, but definitely not two levels.

Also, do we really need/want to expose commands like `stat /run/ostree-booted` to user? They rather seem like implementation details which should rather go/stay in the debug output.

Brainstorm: What about this? If `message` is provided, the command executed is interesting enough for the user to show in the verbose output (level 2), together with the complete command output (level 3).

Reply:
> Brainstorm: What about this? If `message` is provided, the command executed is interesting enough for the user to show in the verbose output (level 2), together with the complete command output (level 3).

And what about doing it the other way around and silent only specific `execute` calls?

Reply:
> In fact, I didn't find any cases where `message` is being provided, are there any?

It seems that only the `podman` plugin uses them:

https://github.com/teemtee/tmt/blob/72701e42fe96a40ce626a8d95fdd7ac9d59a4453/tmt/steps/provision/podman.py#L57-L67

Reply:
> And what about doing it the other way around and silent only specific execute calls?

That also sounds like a reasonable alternative.

Reply:
I have added `silent=False` parameter to `execute()` and used it to silence couple of calls.
I have also decided to preserve `message` since it makes sense to have such possibility. However, I would prefer moving it from `debug` to `verbose` mode.

Reply:
@psss I have changed the indentation in a way that `cmd` is shifted by one step and `out/err` is shifted by two steps (i.e. as it was before). How do you like that?

Reply:
Eventually, I was considering decreasing the indent of `out/err` and align it with `cmd`.

Reply:
Thanks for all the improvements, now looks much better! 

> Eventually, I was considering decreasing the indent of `out/err` and align it with `cmd`.

Yeah, makes sense. The `script` key already clearly shows the next section:

```
    script: dnf install -y tree
        cmd: cd /var/tmp/tmt/run-025/plans/example/tree; export TMT_PLAN_DATA=/var/tmp/tmt/run-025/plans/example/data; set -eo pipefail; dnf install -y tree
        out: Last metadata expiration check: 0:11:16 ago on Fri Nov 25 09:36:37 2022.
        out: Dependencies resolved.
        ...
        out: Installed:
        out:   tree-1.8.0-7.fc35.x86_64                                                      
        out: 
        out: Complete!
    script: ls /
        cmd: cd /var/tmp/tmt/run-025/plans/example/tree; export TMT_PLAN_DATA=/var/tmp/tmt/run-025/plans/example/data; set -eo pipefail; ls /
        out: bin
        out: boot
```

Reply:
Actually, this shifts test output as well. Reverted back.